core/drivers/imx_i2c.c

```diff
@@ -26,10 +26,16 @@
 #define I2C_CLK_CGRBM(__x) 0
 #endif
 
-static struct io_pa_va i2c_bus[] = {
-	{ .pa = I2C1_BASE, },
-	{ .pa = I2C2_BASE, },
-	{ .pa = I2C3_BASE, },
+static struct io_pa_va i2c_bus[3] = {
+#if defined I2C1_BASE
```

Comment:
prefer either `#ifdef I2C1_BASE` or  `#if defined(I2C1_BASE)` for consistency in source files.